### ISSUE TITLE
[Ubuntu] Unclear which file is missing when "Failed to read your VS Code settings file" encountered

### ISSUE THREAD

**Issue by denver-devacademy** [dev-academy-programme/computer-setup#16]
After getting to the end of setup, I noticed "Failed to read your VS Code settings file" popped up.

As far as I can tell, I have executed everything right, however I'm not entirely ruling out having just screwed something up :stuck_out_tongue:

I think it would be handy if there was some kind of check akin to the `type nvm` earlier that the user performs right after creating and modifying the `settings.json` file via VSCode to make sure everything looks right.

Also any help greatly appreciated as I'm still not too sure what I did wrong. I will probably just pull up the code for the checklist package if I can't figure it out in the next few minutes :mag: 

Cheers! :)

**Comment by denver-devacademy** [comment]
Based on taking a peak into the `#computer-setup` channel in the Discord it seems it is for user settings, which on my machine (Pop!_OS) has been dropped at the filepath `~/.config/Code/User/settings.json`

Changing settings via the UI appears to update the file and vice-versa, so going going to assume this one is alright and just an anomaly of the `@donothing/checklist` package :) :+1:

**Comment by gerardpaapu** [comment]
Yeah, sounds like a bug in checklist.

Can you run: `node -e "const os = require('os'); console.log(os.platform(), os.release())"`

These are the values we use to find your settings file and it looks like yours is in the place we expect. It could be a permissions issue

**Comment by polarathene** [comment]
I am going to take a guess that the OS / uname detected was `wsl`? At least it looks like it would never choose `linux` when I looked over the script:

<details>
<summary>Likely caused by a bug in checklist source (Resolved)</summary>

```js
const getUname = async () => {
  const short = await $('uname')
  const full = await $('uname', '-a')

  switch (short.toLowerCase()) {
    case 'linux':
      if (full.toLowerCase().indexOf('microsoft')) {
        return 'wsl'
      }

      return 'linux'

    case 'darwin':
      return 'darwin'

    default:
      throw new Error(`Sorry, I don't know what to do with uname ${short}`)
  }
}
```

`if (full.toLowerCase().indexOf('microsoft'))` would return `-1` since the substring isn't present, but [that will cast to `true`](https://stackoverflow.com/questions/3619797/why-are-javascript-negative-numbers-not-always-true-or-false/3619813#3619813).

That would cause this step to fail:

```js
const findVsCodeUserSettings = async () => {
  switch (await getUname()) {
    case 'darwin':
      return `${OS.homedir()}/Library/Application Support/Code/User/settings.json`

    case 'linux':
      return `${OS.homedir()}/.config/Code/User/settings.json`

    case 'wsl': {
      const APPDATA = await $('wslvar', 'APPDATA')
      const path = await $('wslpath', `${APPDATA}`)

      return `${path}/Code/User/settings.json`
    }
  }
}
```

</details>

Bug was fixed in EDA fork (`@devacademy/checklist`): https://github.com/dev-academy-programme/computer-setup/pull/17#issuecomment-1226646080

**Comment by denver-devacademy** [comment]
> Yeah, sounds like a bug in checklist.
> 
> Can you run: `node -e "const os = require('os'); console.log(os.platform(), os.release())"`
> 
> These are the values we use to find your settings file and it looks like yours is in the place we expect. It could be a permissions issue

@gerardpaapu No worries, here's the output :)

```bash
╭─denver-da@pop-os ~ 
╰─$ node -e "const os = require('os'); console.log(os.platform(), os.release())"
linux 5.18.10-76051810-generic
```

**Comment by polarathene** [comment]
Try `npx --yes @devacademy/checklist`.

**Comment by gerardpaapu** [comment]
Yeah, the computer-setup instructions need to be updated. I moved checklist from the `@donothing` org to the `@devacademy` org and then fixed that bug (or Elise did I think).

**Comment by humaid-memee** [comment]
I'm also getting the following "Failed to read your VS Code settings file" after running "npx --yes @devacademy/checklist"

I have tried to run " node -e "const os = require('os'); console.log(os.platform(), os.release())" ", and got the below output:

![image](https://github.com/dev-academy-programme/computer-setup/assets/130061205/7c97d9d3-28ed-4aab-a587-a12b4eb1841f)

**Comment by gerardpaapu** [comment]
This is where we look for your settings file:

```sh
$ echo "$(wslpath $(wslvar APPDATA))/Code/User/settings.json"
/mnt/c/Users/gerard/AppData/Roaming/Code/User/settings.json

$ cat "$(wslpath $(wslvar APPDATA))/Code/User/settings.json"
{ ... }
```

Can you tell me if your settings file is there, and if not there, where it is? Thanks.

**Comment by gerardpaapu** [comment]
Oh, and if you're just worried about completing your setup, we know that this last check is a bit flaky on Windows so you can just get your facilitator to visually double check your settings file

**Comment by humaid-memee** [comment]
I have a settings file where you've mentioned, but another one here
C:\Users\memee\AppData\Local\Packages\Microsoft.WindowsTerminal_8wekyb3d8bbwe\LocalState\settings.json

Thanks Gerard, I'll check in with them in on Monday.
